pyathena/converter.py

```diff
@@ -27,6 +29,13 @@ def _to_datetime(varchar_value: Optional[str]) -> Optional[datetime]:
     return datetime.strptime(varchar_value, "%Y-%m-%d %H:%M:%S.%f")
 
 
+def _to_datetime_with_tz(varchar_value: Optional[str]) -> Optional[datetime]:
+    if varchar_value is None:
+        return None
+    datetime_, _, tz = varchar_value.rpartition(" ")
+    return datetime.strptime(datetime_, "%Y-%m-%d %H:%M:%S.%f").replace(tzinfo=gettz(tz))
```

Comment:
`from dateutil.tz import gettz`
If you use dateutil, `dateutil.parser.parse` will eliminate the need to replace timezone.

Reply:
That's the first thing I tried, thinking it would work out of the box. But `dateutil.parser.parse` doesn't parse IANA style timezones. Running:
```python
from pyathena import connect

cursor = connect(
    s3_staging_dir="s3://tendam-temporary/dbt-dev", region_name="eu-west-1"
).cursor()
cursor.execute("select current_timestamp(6) at time zone 'Europe/Madrid'")
cursor.fetchall()
```
results in

```shell
Traceback (most recent call last):
  File "/Users/49353/PyAthena/test_converter_timestamp_with_tz.py", line 8, in <module>
    cursor.execute("select current_timestamp(6) at time zone 'Europe/Madrid'")
  File "/Users/49353/PyAthena/pyathena/cursor.py", line 100, in execute
    self.result_set = self._result_set_class(
                      ^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/49353/PyAthena/pyathena/result_set.py", line 63, in __init__
    self._pre_fetch()
  File "/Users/49353/PyAthena/pyathena/result_set.py", line 324, in _pre_fetch
    self._process_rows(response)
  File "/Users/49353/PyAthena/pyathena/result_set.py", line 417, in _process_rows
    processed_rows = self._get_rows(offset, metadata, rows)
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/49353/PyAthena/pyathena/result_set.py", line 399, in _get_rows
    self._converter.convert(meta.get("Type"), row.get("VarCharValue"))
  File "/Users/49353/PyAthena/pyathena/converter.py", line 157, in convert
    return converter(value)
           ^^^^^^^^^^^^^^^^
  File "/Users/49353/PyAthena/pyathena/converter.py", line 35, in _to_datetime_with_tz
    return parse(varchar_value)
           ^^^^^^^^^^^^^^^^^^^^
  File "/Users/49353/PyAthena/.venv/lib/python3.12/site-packages/dateutil/parser/_parser.py", line 1368, in parse
    return DEFAULTPARSER.parse(timestr, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/49353/PyAthena/.venv/lib/python3.12/site-packages/dateutil/parser/_parser.py", line 643, in parse
    raise ParserError("Unknown string format: %s", timestr)
dateutil.parser._parser.ParserError: Unknown string format: 2024-08-02 11:28:36.771074 Europe/Madrid
```

Reply:
I see, then I will check to see if the test passes.